bdns/dns.go

```diff
@@ -475,3 +488,60 @@ func (dnsClient *DNSClientImpl) LookupCAA(ctx context.Context, hostname string)
 	}
 	return CAAs, nil
 }
+
+// logDNSError logs the provided err result from making a query for hostname. If
+// the err is a `dns.ErrId` instance then the raw Base64 URL encoded bytes of
+// the query (and if not-nil, the response) in wire format is logged as well.
+// This function is called from exchangeOne only for the case where an error
+// occurs querying a hostname that indicates a problem between the VA and the
+// resolver.
+func (dnsClient *DNSClientImpl) logDNSError(hostname string, msg, resp *dns.Msg, err error) {
+	// We use a stand-alone function for this to make it easy to share the
+	// implementation between the DNSClientImpl and MockDNSClient
+	logDNSError(dnsClient.log, hostname, msg, resp, err)
+}
+
+// log a DNS level error with the given logger. See DNSClientImpl.logDNSError.
+func logDNSError(logger blog.Logger, hostname string, msg, resp *dns.Msg, underlying error) {
```

Comment:
Might also want to include `chosenServer` in these messages.

Reply:
Good call :+1:

Reply:
Added in 705bd19 Thanks!